lms/djangoapps/instructor/views/api.py

```diff
@@ -2437,6 +2441,353 @@ def problem_grade_report(request, course_id):
+        error_msg = _("Failed to communicate with gradebook server at {url}").format(url=rgburl) + "<br/>"
+        error_msg += _("Error: {err}").format(err=err)
+        error_msg += "<br/>resp={resp}".format(resp=resp.content)
+        error_msg += "<br/>data={data}".format(data=data)
+        return error_msg, {}
+
+    retdata = retdict['data']  # a list of dicts
+    if retdata:
+        datatable = {'header': retdata[0].keys()}
+        datatable['data'] = [x.values() for x in retdata]
+        datatable['retdata'] = retdata
+    else:
+        datatable = {}
+
+    return None, datatable
+
+
+@require_POST
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('staff')
+def list_remote_assignments(request, course_id):
+    """
+    Returns a datatable of the assignments available in the remote gradebook
+    """
+    course_id = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    course = get_course_by_id(course_id)
+    error_msg, datatable = _do_remote_gradebook(request.user, course, 'get-assignments')
+    datatable['title'] = _('Remote Gradebook Assignments')
+    return JsonResponse({
+        'errors': error_msg,
+        'datatable': datatable,
+    })
+
+
+@require_POST
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('staff')
+def list_remote_enrolled_students(request, course_id):
+    """
+    Returns a datatable of students and whether or not there is a match for those students
+    in the remote gradebook
+    """
+    course_id = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    course = get_course_by_id(course_id)
+    student_data = get_student_grade_summary_data(course, get_grades=False)
+    error_msg, rg_student_data = _do_remote_gradebook(request.user, course, 'get-membership')
+    datatable = {'header': ['Student  email', 'Match?']}
+    rg_students = [x['email'] for x in rg_student_data['retdata']]
+
+    def domatch(student):
+        """Returns 'yes' if student is pressent in the remote gradebook student list, else returns 'No'"""
+        return 'yes' if student.email in rg_students else 'No'
+
+    datatable['data'] = [[x.email, domatch(x)] for x in student_data['students']]
+    datatable['title'] = _('Enrolled Students Matching Remote Gradebook')
+    return JsonResponse({
+        'errors': error_msg,
+        'datatable': datatable
+    })
+
+
+@require_POST
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('staff')
+def list_course_assignments(request, course_id):
+    """
+    Returns a datatable of the assignments available for this course
+    """
+    course_id = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    course = get_course_by_id(course_id)
+    allgrades = get_student_grade_summary_data(course, get_grades=True)
+    assignments = [[x] for x in allgrades['assignments']]
```

Comment:
why each assignment in array , i mean why 2d array, is `x` is it self array or single object.